src/components/Navbar/index.spec.tsx

```diff
@@ -1,69 +1,104 @@
 
 import Navbar from '.';
 
 describe(__filename, () => {
-  const render = (props = {}) => {
-    return shallow(<Navbar onLogOut={() => null} profile={null} {...props} />);
+  type RenderParams = {
+    _logOutFromServer?: typeof logOutFromServer;
+    store?: Store;
   };
 
-  const createProfile = (name = 'Bob') => {
-    return { name };
+  const render = ({
+    _logOutFromServer = jest.fn(),
+    store = configureStore(),
+  }: RenderParams = {}) => {
+    // TODO: Use shallowUntilTarget()
+    // https://github.com/mozilla/addons-code-manager/issues/15
+    const root = shallow(<Navbar _logOutFromServer={_logOutFromServer} />, {
+      context: { store },
+    }).shallow();
+
+    return root;
+  };
+
+  const storeWithUser = (user = fakeUser) => {
+    const store = configureStore();
+
+    store.dispatch(userActions.loadCurrentUser({ user }));
+    return store;
   };
 
   describe('when a profile is provided', () => {
     it('renders a username', () => {
       const name = 'Bob';
-      const root = render({ profile: createProfile(name) });
+      const store = storeWithUser({ ...fakeUser, name });
+      const root = render({ store });
 
       expect(root.find(`.${styles.username}`)).toHaveText(name);
     });
 
     it('renders a log out button', () => {
-      const onLogOut = () => null;
-      const root = render({ onLogOut, profile: createProfile() });
+      const root = render({ store: storeWithUser() });
 
       expect(root.find(`.${styles.logOut}`)).toHaveLength(1);
     });
 
     it('does not render a log in button', () => {
-      const root = render({ profile: createProfile() });
+      const root = render({ store: storeWithUser() });
 
       expect(root.find(LoginButton)).toHaveLength(0);
     });
   });
 
   describe('when profile is null', () => {
     it('does not render a username', () => {
-      const root = render({ profile: null });
+      const root = render();
 
       expect(root.find(`.${styles.username}`)).toHaveLength(0);
     });
 
     it('renders a log in button', () => {
-      const root = render({ profile: null });
+      const root = render();
 
       expect(root.find(LoginButton)).toHaveLength(1);
     });
 
     it('does not render a log out button', () => {
-      const root = render({ profile: null });
+      const root = render();
 
       expect(root.find(`.${styles.logOut}`)).toHaveLength(0);
     });
   });
 
   describe('Log out button', () => {
-    it('calls onLogOut when clicked', () => {
-      const onLogOutMock = jest.fn();
-      const root = render({ onLogOut: onLogOutMock, profile: createProfile() });
+    it('calls logOutFromServer when clicked', () => {
+      const logOutFromServerMock = jest.fn();
+      const root = render({
+        _logOutFromServer: logOutFromServerMock,
+        store: storeWithUser(),
+      });
+
+      root.find(`.${styles.logOut}`).simulate('click');
+      expect(logOutFromServerMock).toHaveBeenCalled();
```

Comment:
I haven't tried running the code but `this.logOut()` on the class is async so that's probably the problem. We rarely had to deal with this on `addons-frontend` because sagas were such a testable abstraction. 

Until we have a testable abstraction (which hopefully will happen in https://github.com/mozilla/addons-code-manager/issues/12), you could split it into two tests. One that verifies `this.logOut()` gets called and another that calls `root.instance().logOut()` directly so that it can wait for the promise before making assertions.

Reply:
Can't we `await` the call above? I think we can do things like this because Jest definitely supports async code.

Reply:
That works @willdurand, thanks!